lisa/sut_orchestrator/azure/tools.py

```diff
@@ -139,6 +140,47 @@ def get_distro_version(self) -> str:
             if python_exists:
                 break
 
+        self._python_cmd = python_cmd
+        self._python_use_sudo = use_sudo
+
+        return self._python_cmd, self._python_use_sudo
+
+    def get_waagent_conf_path(self) -> str:
+        if self._waagent_conf_path is not None:
+            return self._waagent_conf_path
+
+        python_cmd, use_sudo = self.get_python_cmd()
+
+        # Try to use waagent code to detect
+        result = self.node.execute(
+            f'{python_cmd} -c "from azurelinuxagent.common.osutil import get_osutil;'
```

Comment:
Not sure if this is an issue, just FYI. For the import to work, python_cmd needs to be exactly the same Python used to run the agent (otherwise the azurelinuxagent module won't be installed). Some distros, RedHat was one of them, started using a "platform python" that was used for the agent and some other utilities and included a limited set of modules. Then users could install other Python versions on their own.

Reply:
also just FYI: to workaround those issues, our automation gets the python that is running the agent from the /proc filesystem. That works if the Agent is currently running; otherwise we have a fallback logic that tries to discover the correct Python. The full logic is here: https://github.com/Azure/WALinuxAgent/blob/develop/tests_e2e/orchestrator/scripts/get-agent-python